features/bsm/daemon/src/main/java/org/opennms/netmgt/bsm/daemon/Bsmd.java

```diff
@@ -0,0 +1,310 @@
+    }
+
+    @EventHandler(uei = EventConstants.RELOAD_DAEMON_CONFIG_UEI)
+    public void handleReloadEvent(Event e) {
+        LOG.info("Received a reload configuration event: {}", e);
+
+        final Parm daemonNameParm = e.getParm(EventConstants.PARM_DAEMON_NAME);
+        if (daemonNameParm == null || daemonNameParm.getValue() == null) {
+            LOG.warn("The {} parameter has no value. Ignoring.", EventConstants.PARM_DAEMON_NAME);
+            return;
+        }
+
+        if (NAME.equalsIgnoreCase(daemonNameParm.getValue().getContent())) {
+            LOG.info("Reloading bsmd.");
+
+            EventBuilder ebldr = null;
+            try {
+                handleConfigurationChanged();
+
+                ebldr = new EventBuilder(EventConstants.RELOAD_DAEMON_CONFIG_SUCCESSFUL_UEI, NAME);
+                ebldr.addParam(EventConstants.PARM_DAEMON_NAME, NAME);
+                LOG.info("Reload succesful.");
+            } catch (Throwable t) {
+                ebldr = new EventBuilder(EventConstants.RELOAD_DAEMON_CONFIG_FAILED_UEI, NAME);
+                ebldr.addParam(EventConstants.PARM_REASON, t.getLocalizedMessage().substring(0, 128));
+                LOG.error("Reload failed.", t);
+            }
+
+            ebldr.addParam(EventConstants.PARM_DAEMON_NAME, NAME);
+            m_eventIpcManager.sendNow(ebldr.getEvent());
+        }
+    }
+
+    @Override
+    public void destroy() throws Exception {
+        LOG.info("Stopping bsmd...");
+    }
+
+    public void setBusinessServiceDao(BusinessServiceDao businessServiceDao) {
+        m_businessServiceDao = businessServiceDao;
+    }
+
+    public BusinessServiceDao getBusinessServiceDao() {
+        return m_businessServiceDao;
+    }
+
+    public void setAlarmDao(AlarmDao alarmDao) {
+        m_alarmDao = alarmDao;
+    }
+
+    public AlarmDao getAlarmDao() {
+        return m_alarmDao;
+    }
+
+    public void setEventIpcManager(EventIpcManager eventIpcManager) {
+        m_eventIpcManager = eventIpcManager;
+    }
+
+    public EventIpcManager getEventIpcManager() {
+        return m_eventIpcManager;
+    }
+
+    public void setEventConfDao(EventConfDao eventConfDao) {
+        m_eventConfDao = eventConfDao;
+    }
+
+    public EventConfDao getEventConfDao() {
+        return m_eventConfDao;
+    }
+
+    public void setTransactionTemplate(TransactionTemplate template) {
+        m_template = template;
+    }
+
+    public TransactionTemplate getTransactionTemplate() {
+        return m_template;
+    }
+
+    public void setVerifyReductionKeys(boolean verify) {
+        m_verifyReductionKeys = verify;
+    }
+
+    public boolean getVerifyReductionKeys() {
+        return m_verifyReductionKeys;
+    }
+
+    public void setBusinessServiceStateMachine(BusinessServiceStateMachine stateMachine) {
+        m_stateMachine = stateMachine;
+    }
+
+    public BusinessServiceStateMachine getBusinessServiceStateMachine() {
+        return m_stateMachine;
+    }
+}
```

Comment:
Are the getters and setters actively used somewhere? 
The Bsmd should never be used manually and therefore should not expose anything publicly.